Sdks.bee.cs

```diff
@@ -0,0 +1,166 @@
+            // libraries (Enlighten), compilation with newer standard libraries doesn't work, and is not trivial
+            // to fix. So for MSVC we will only use the local install if it is exactly the right version, and
+            // will ignore any newer toolchains.
+            candidates = candidates.Where(sdk => sdk.MsvcToolsetVersion.Major == msvcToolsetVersion.Major
+                && sdk.MsvcToolsetVersion.Minor == msvcToolsetVersion.Minor
+                // Comparing the 'Build' version is necessary because 14.28 had
+                // breaking changes between sub-minor release numbers
+                && sdk.MsvcToolsetVersion.Build == msvcToolsetVersion.Build);
+        }
+
+        // Prefer locally installed SDKs over downloadable ones
+        candidates = candidates.OrderBy(sdk => sdk.IsDownloadable ? 1 : 0)
+            .ThenByDescending(sdk => sdk.MsvcVersion)
+            .ThenByDescending(sdk => sdk.Win10SdkVersion);
+
+        var locatedSdk = candidates.DefaultIfEmpty(null).FirstOrDefault();
+        if (locatedSdk == null)
+        {
+            var hashKey = new ValueTuple<SdkLocator<T>, Version, Version>(locator, win10SdkVersion, msvcToolsetVersion);
+            locatedSdk = Backend.Current.ExecuteOnce(hashKey, () =>
+            {
+                var message = "No Visual Studio SDK ";
+                if (msvcToolsetVersion != null)
+                {
+                    message += $"with MSVC toolset version {msvcToolsetVersion.Major}.{msvcToolsetVersion.Minor} ";
+                    if (win10SdkVersion != null)
+                        message += "and ";
+                }
+
+                if (win10SdkVersion != null)
+                    message += $"Win10SDK version {win10SdkVersion} ";
+
+                message += "could be found.\n";
+
+                T resultSdk = locator.UserDefault;
+                message += resultSdk != null
+                    ? $"The UserDefault SDK with MSVC toolset version {resultSdk.MsvcToolsetVersion} and Win10 SDK version {resultSdk.Win10SdkVersion} will be used; be aware that this may cause build issues.\n"
+                    : "No default VisualStudio SDK was found on this machine.\n";
+
+                message += "\nFor reference, the following MSVC toolset versions are available:\n";
+                message += locator.All.Exclude(locator.Dummy).Select(sdk => $"\t{sdk.MsvcToolsetVersion} ({(sdk.IsDownloadable ? "online" : "local")})").Distinct().SeparateWith("\n");
+                message += "\n\nand the following Win10 SDK versions are available:\n";
+                message += locator.All.Exclude(locator.Dummy).Select(sdk => $"\t{sdk.Win10SdkVersion} ({(sdk.IsDownloadable ? "online" : "local")})").Distinct().SeparateWith("\n");
+                message += "\n";
+
+                Errors.PrintWarning(message);
+                return resultSdk;
+            });
+        }
+
+        return locatedSdk ?? locator.Dummy;
```

Comment:
Do you intend this to be buildable by anyone who finds this repo on github? If not, none of this is necessary.